lib/jormungandr/src/Cardano/Wallet/Jormungandr/Rewards.hs

```diff
@@ -0,0 +1,127 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE TypeApplications #-}
+module Cardano.Wallet.Jormungandr.Rewards
+    (
+      rankingEpochConstants
+    , rewardParamsFromBlock0
+    , calculateRewards
+    , applyLimit
+    , rewardsAt
+    )
+    where
+
+import Prelude
+
+import Cardano.Wallet.Primitive.Types
+    ( EpochNo (..) )
+import Data.Maybe
+    ( mapMaybe )
+import Data.Quantity
+    ( Quantity (..) )
+import Data.Word
+    ( Word64 )
+
+import qualified Cardano.Pool.Ranking as Ranking
+import Cardano.Wallet.Jormungandr.Binary
+
+-- Rewards
+--
+-- Based on the information in:
+-- https://github.com/input-output-hk/chain-libs/blob/master/chain-impl-mockchain/doc/incentives.md
+
+rewardParamsFromBlock0
+    :: Block
+    -> Maybe (RewardFormula, TreasuryTax, Maybe Ratio)
+rewardParamsFromBlock0 block = do
+        rewardParams <- find $ \case
+                ConfigRewardFormula x -> Just x
+                _ -> error "missing config reward"
```

Comment:
Why use `find` and `Maybe` for safety while using `error` for missing values?

Reply:
Oops, this was while debugging, to have more information than `Maybe`. Ultimately an `Either` would be good, I think, but I am not prioritising that.